src/Core/Content/Shared/MailFlow/CustomerCriteriaBuilder.php

```diff
@@ -0,0 +1,31 @@
+<?php
+declare(strict_types=1);
+
+namespace Shopware\Core\Content\Shared\MailFlow;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+
+/**
+ * @internal
+ */
+#[Package('after-sales')]
+class CustomerCriteriaBuilder
+{
+    public function getCriteria(string $entityId): Criteria
```

Comment:
Just an idea: this would also fit as a static function inside `CustomerProvider` or? Because it's so simple and primarily used there (can still be public to be used elsewhere)